programs/launcher/main.cpp

```diff
@@ -402,19 +434,53 @@ launcher_def::initialize (const variables_map &vmap) {
   alias_base = "testnet_";
   next_node = 0;
 
+  load_servers ();
+
   if (prod_nodes > producers)
     prod_nodes = producers;
   if (prod_nodes > total_nodes)
     total_nodes = prod_nodes;
 
-  if (host_map.bindings.empty()) {
-    define_local ();
+  if (bindings.empty()) {
+    define_network ();
   }
 }
 
+void
+launcher_def::load_servers () {
+  if (!server_ident_file.empty()) {
+    try {
+      fc::json::from_file(server_ident_file).as<server_identities>(servers);
+      size_t nodes = 0;
+      for (auto &s : servers.producer) {
+        nodes += s.instances;
+      }
+      prod_nodes = nodes;
+      nodes = 0;
+      for (auto &s : servers.observer) {
+        nodes += s.instances;
+      }
+
+      total_nodes = prod_nodes + nodes;
+      per_host = 1;
+      network.ssh_helper = servers.ssh;
+    }
+    catch (const exception &ex) {
+      elog( "caught assert on fetch_block_by_id, ${ex}",("ex",ex.what()));
```

Comment:
Looks like a copy/paste error. Also please use explicit `std::exception` here as `fc::exception` and `std::exception` are both in play and it is not immediately obvious which this is. Note that `fc::exception` does not derive from `std::exception` which means we normally need a catch for each one.

Reply:
this was a copy/paste, but the error was not pulling it before committing. At this moment I would prefer to stick with the minimalist approach to error processing. We can switch to more rigorous error handling later, but this is a command-line utility that does its business and then terminates, so I'd rather make enhanced error reporting a post-12/4 activity.